File: In_C_Programming/q8.c
#include <stdio.h>

void sort_Array(int arr[], int ln)
{

    for (int i = 0; i < ln; i++)
    {
        for (int j = ln - 1; j > i; j--)
        {

            if (arr[i] > arr[j])
            {
                arr[i] ^= arr[j];
                arr[j] ^= arr[i];
                arr[i] ^= arr[j];
            }
        }
    }
}

int Order_Search(int arr[], int ln, int target)
{

    sortArray(arr, ln);

    for (int i = 0; i < ln; i++)
    {
        if (arr[i] == target)
        {
            return i;
        }
        else if (arr[i] > target)
        {
            return -1;
        }
    }

    return -1;
}

int Unorder_Search(int arr[], int ln, int target)
{

    for (int i = 0; i < ln; i++)
    {
        if (arr[i] == target)
        {
            return i;
        }
    }

    return -1;
}

int main()
{

    int size, target;

    printf("Enter THe Array Length: \n");
    scanf("%d", &size);

    int arr[size];

    printf("Enter The Elements: \n");

    for (int i = 0; i < size; i++)
    {
        scanf("%d", &arr[i]);
    }

    printf("Enter The Target: ");
    scanf("%d", &target);

    int result1 = Unorder_Search(arr, size, target);
    int result2 = Order_Search(arr, size, target);

    if (result1 != -1 && result2 != -1)
    {
        printf("Value Found in %d Position in Un-Ordered Linear Search \n", result1 + 1);
        printf("Value Found in %d Position in Ordered Linear Search \n", result2 + 1);
    }
    else
    {
        printf("Value not Found!");
    }

    return 0;
}

File: In_Java_Programming/q8.java
import java.util.Scanner;

public class q8 {

    public static int linearSearchOrder(int arr[], int ln, int target) {

        for (int i = 0; i < ln - 1; i++) {

            for (int j = 0; j < ln - 1 - i; j++) {

                if (arr[j] > arr[j + 1]) {

                    arr[j] ^= arr[j + 1];
                    arr[j + 1] ^= arr[j];
                    arr[j] ^= arr[j + 1];
                }
            }
        }

        for (int i = 0; i < ln; i++) {

            if (target == arr[i]) {

                return i + 1;

            }

            if (target < arr[i]) {

                return -1;
            }
        }

        return -1;
    }

    public static int linearSearchUnOrder(int arr[], int ln, int target) {

        for (int i = 0; i < ln; i++) {

            if (target == arr[i]) {

                return i + 1;
            }
        }

        return -1;
    }

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        System.out.println("Enter The Array Length: ");
        int size = sc.nextInt();

        int arr[] = new int[size];

        System.out.println("Enter The Elements: ");

        for (int i = 0; i < size; i++) {

            arr[i] = sc.nextInt();
        }

        System.out.println("Enter The Key Element: ");
        int key = sc.nextInt();

        int result = linearSearchOrder(arr, size, key);
        int result2 = linearSearchUnOrder(arr, size, key);

        if (result == -1 && result2 == -1) {

            System.out.println("Element not Found on Both Array!");
        } else {

            System.out.println("Element Found on " + result + " Position on Ordered Array");
            System.out.println("Element Found on " + result2 + " Position  on Un-Ordered Array");
        }

        sc.close();

    }
}